cnf-certification-test/lifecycle/suite.go

```diff
@@ -555,3 +565,24 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
+func testStorageRequiredPods(env *provider.TestEnvironment) {
+	oc := clientsholder.GetClientsHolder()
+	var nonCompliantPods []*provider.Pod
+	for _, pod := range env.Pods {
+		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
+			if podOwner.Kind != statefulSet {
+				continue
+			}
+			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
+			if err != nil {
+				tnf.ClaimFilePrintf(err.Error())
+				continue
+			}
+			if statefulset.Spec.ServiceName == localStorage {
```

Comment:
I think what you meant to check is the `storageClassName`, part of the `volumeClaimTemplates` section of the `StatefulSet`. In any case, I'm not sure if this is the right way to verify this requirement because reading the documentation for [local persistent volumes](https://kubernetes.io/blog/2019/04/04/kubernetes-1.14-local-persistent-volumes-ga/), that "local-storage" seems to be just a convenient name for the `StorageClass`, so any other could be used instead.

Besides that, the PVCs representing local storage can be attached not only to "StatefulSets" but also to "Pods" and maybe other resources.